src/registrar/models/user.py

```diff
@@ -169,6 +247,49 @@ def get_formatted_name(self):
     def has_contact_info(self):
         return bool(self.title or self.email or self.phone)
 
+    def _get_portfolio_permissions(self):
+        """
+        Retrieve the permissions for the user's portfolio roles.
+        """
+        portfolio_permissions = set()  # Use a set to avoid duplicate permissions
+
+        if self.portfolio_roles:
+            for role in self.portfolio_roles:
+                if role in self.PORTFOLIO_ROLE_PERMISSIONS:
+                    portfolio_permissions.update(self.PORTFOLIO_ROLE_PERMISSIONS[role])
+        if self.portfolio_additional_permissions:
+            portfolio_permissions.update(self.portfolio_additional_permissions)
+        return list(portfolio_permissions)  # Convert back to list if necessary
+
+    def _has_portfolio_permission(self, portfolio_permission):
+        """The views should only call this guy when testing for perms and not rely on roles"""
```

Comment:
**suggestion** "this guy" 👀. Maybe remove that. Also, are views then expected to pass in things like "view_permission" etc? if so I would suggest moving the permissions enum to a utility to avoid any chance of circular imports down the road with multiple files using those enums by importing from user.

Reply:
I like the idea of moving the enum to a utility but I don't get how we're risking circular imports (right now a view would import from user for both the constants and the function. The function refers to its class. A circular dependency would imply that something in user would need to import from the view, no?)

Reply:
In other words, isn't this a similar architecture to how we're handling domain state and request status, where these enums are a class prop and there are methods that we pass these to?

Reply:
i agree with rachid.  i think we are following de facto standard established in these other models